src/errors_handler.rs

```diff
@@ -0,0 +1,492 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Keylime Authors
+
+use crate::common::{APIVersion, JsonWrapper, API_VERSION};
+use actix_web::{
+    body::ResponseBody,
+    dev,
+    error::{JsonPayloadError, PathError, QueryPayloadError},
+    http,
+    middleware::errhandlers::{ErrorHandlerResponse, ErrorHandlers},
+    web, Error, HttpRequest, HttpResponse, Responder, Result,
+};
+use log::*;
+
+pub(crate) async fn app_default(req: HttpRequest) -> impl Responder {
+    let error;
+    let response;
+    let message;
+
+    match req.head().method {
+        http::Method::GET => {
+            error = 400;
+            message = format!(
+                "Not Implemented: Use /version or /{}/ interfaces",
+                API_VERSION
+            );
+            response = HttpResponse::BadRequest()
+                .json(JsonWrapper::error(error, &message));
+        }
+        http::Method::POST => {
```

Comment:
I was going to suggest combining this with the GET match but there is a slight difference in the message.